### PR TITLE
kvclientv2: fix resolvedts go back

### LINKED ISSUE #1548: panic in cdc server: The CRTs must be greater than the resolvedTs
## Bug Report

Please answer these questions before submitting your issue. Thanks!
```
[2021/03/25 09:28:41.983 +08:00] [PANIC] [puller.go:194] ["The CRTs must be greater than the resolvedTs"] [row="{\"OpType\":3,\"Key\":null,\"Value\":null,\"OldValue\":null,\"StartTs\":0,\"CRTs\":423791048719335480,\"RegionID\":293891616}"] [CRTs=423791048719335480] [resolvedTs=4237
91347484852344] [tableID=1364] [stack="github.com/pingcap/ticdc/cdc/puller.(*pullerImpl).Run.func5.1\n\tgithub.com/pingcap/ticdc@/cdc/puller/puller.go:194\ngithub.com/pingcap/ticdc/cdc/puller.(*pullerImpl).Run.func5\n\tgithub.com/pingcap/ticdc@/cdc/puller/puller.go:253\ngolang.org/
x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
```

logs:
http://172.16.5.101/vF4is/626.tgz
http://172.16.5.101/zScQd/625.tgz
http://172.16.5.101/4lWbf/627.tgz

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 

fix #1548

when a stream to kvstore is closed, kvclientv2 sent all region info under this stream with `onRegionFail`.
but the checkpoint is not updated in this time.

### Check List 

Tests 

 - Unit test
### Release note

- No release note (this bug is never released)

### CODE REVIEW

File: cdc/kv/client_test.go [hunk 1]
```diff
@@ -1300,6 +1301,144 @@ func (s *etcdSuite) testStreamRecvWithError(c *check.C, failpointStr string) {
...
+		{
+			Resolved: &model.ResolvedSpan{
+				Span:       regionspan.ComparableSpan{Start: []byte("a"), End: []byte("b")},
+				ResolvedTs: 120,
+			},
+			RegionID: regionID,
+		},
+	}
+
+	defer cancel()
```
Comment (leoppro) [inline cdc/kv/client_test.go:1428]:
```suggestion
```

File: cdc/kv/client_test.go [hunk 2]
```diff
@@ -1300,6 +1301,143 @@ func (s *etcdSuite) testStreamRecvWithError(c *check.C, failpointStr string) {
...
+				ResolvedTs: 120,
+			},
+			RegionID: regionID,
+		},
+	}
+
+	for _, expectedEv := range expected {
+		select {
+		case event := <-eventCh:
+			log.Info("receive event", zap.Reflect("event", event), zap.Reflect("expected", expectedEv))
```
Comment (amyangfei) [inline cdc/kv/client_test.go:1431]:
```suggestion
```

File: pkg/util/failpoint.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/pingcap/failpoint"
+)
+
+// FailpointBuild is ture if this is a failpoint build
+var FailpointBuild = isFailpointBuild()
+
+// using failpoint package when the failpoint is enabled to avoid imported and not used error
+var _failpointValue = failpoint.Value(0) //nolint
+
+func isFailpointBuild() bool {
```
Comment (amyangfei) [inline pkg/util/failpoint.go:26]: please add a unit test for this function

**Pull request merged**